packages/docs-infra/src/CodeHighlighter/types.ts

```diff
@@ -0,0 +1,170 @@
+import type { Root, RootData } from 'hast';
+import type { Delta } from 'jsondiffpatch';
+
+export type Components = { [key: string]: React.ReactNode };
+
+type CodeMeta = {
+  fileName?: string;
+  path?: string;
+};
+
+export type Transforms = Record<string, { delta: Delta; fileName?: string }>;
+
+// External import definition matching parseImports.ts
+export interface ExternalImportItem {
+  name: string;
+  type: 'named' | 'default' | 'namespace';
+  isType?: boolean;
+}
+
+export type Externals = Record<string, ExternalImportItem[]>;
+
+export interface HastRoot extends Root {
+  data?: RootData & { totalLines?: number };
+}
+
+export type VariantSource = string | HastRoot | { hastJson: string };
+export type VariantExtraFiles = {
+  [fileName: string]:
+    | string
+    | {
+        source?: VariantSource;
+        transforms?: Transforms;
+        skipTransforms?: boolean;
+        metadata?: boolean;
+        path?: string;
+      };
+};
+export type VariantCode = CodeMeta & {
+  url?: string;
+  source?: VariantSource;
+  extraFiles?: VariantExtraFiles;
+  metadataPrefix?: string;
+  externals?: string[];
+  namedExport?: string;
+  filesOrder?: string[];
+  transforms?: Transforms;
+  allFilesListed?: boolean;
+  skipTransforms?: boolean;
+};
+export type Code = { [key: string]: undefined | string | VariantCode }; // TODO: only preload should be able to pass highlighted code
+
+export type ControlledVariantExtraFiles = {
+  [fileName: string]: { source: string | null };
+};
+export type ControlledVariantCode = CodeMeta & {
+  url?: string;
+  source?: string | null;
+  extraFiles?: ControlledVariantExtraFiles;
+  filesOrder?: string[];
+};
+export type ControlledCode = { [key: string]: undefined | null | ControlledVariantCode };
+
+type BaseContentProps = {
+  name?: string;
+  slug?: string;
+  code?: Code;
+  components?: Components;
+  url?: string;
+  variantType?: string;
+};
+
+export type ContentProps<T extends {}> = BaseContentProps & T;
+export type ContentLoadingVariant = {
+  fileNames?: string[];
+  source?: React.ReactNode;
+  extraSource?: { [fileName: string]: React.ReactNode };
+};
+export type BaseContentLoadingProps = ContentLoadingVariant & {
+  extraVariants?: Record<string, ContentLoadingVariant>;
+  name?: string;
+  slug?: string;
+  url?: string;
+};
+export type ContentLoadingProps<T extends {}> = BaseContentLoadingProps & T;
+
+type ErrorHandler = React.ComponentType<{ error: Error }>;
+
+interface CodeHighlighterBaseProps {
+  name?: string;
```

Comment:
Perhaps add some jsdoc annotations? It could save us from writing and maintaining some of the documentation if code would self-document what it can through IDE code hints.